### DISCUSSION TITLE
Support for Read-Multiple-Variable-Length-Characteristic-Values

### DISCUSSION THREAD

**Discussion by GardeningStevie** [hbldh/bleak#1669] [Ideas]
Hi Bleak,

I have some BLE devices that have many small characteristics and the read performance is poor if `read_gatt_char` does not support reading multiple charateristics at once.

So, is it possible to extend `read_gatt_char` to accept a list of characteristic handles which are passed as `ATT_READ_MULTIPLE_VARIABLE_REQ` command to the Bluetooth backends?

**Comment by dlech** [comment]
No, this is something that would have to be added at the OS level on all platforms first.

There are some USB Bluetooth dongles that bypass the OS Bluetooth stack that would allow making use of low-level commands like this, but they aren't supported with Bleak.

**Reply by GardeningStevie** [reply]
So there is no OS platform that supports a feature that was introduced a decade ago?
Frustrating.

I looked at the Android and iOS API and couldn't find any proof of existence. I could not believe that this is true...
